### DISCUSSION TITLE
Semicolons should not be put on the next line

### DISCUSSION THREAD

**Discussion by nfcampos** [tconbeer/sqlfmt#278] [The sqlfmt Style]
**Describe the bug**
Semicolons should not be put on the next line

**To Reproduce**
```
select * from revenue;
```

**Expected behavior**
```
select *
from revenue;
```

**Actual behavior**
```
select *
from revenue
;
```

**Additional context**
What is the output of `sqlfmt --version`?
```
sqlfmt, version 0.11.1
```

**Comment by tconbeer** [comment]
This is an intentional part of the style today. I like the visual separator when there are multiple queries in one file:

```sql
select *
from revenue
;

select *
from costs
;
```

(Unlike a language like javascript where a semicolon terminates each statement, here it terminates the whole query, so I feel that it should be at least as prominent as the main keywords in that query.)

What are the arguments for merging it with the last line?

**Reply by elyobo** [reply]
It's adding a lot of noise to a simple test file (supabase, pgtap) which inherently has many statements.

```sql
create extension if not exists "basejump-supabase_test_helpers";

begin
;

select plan(1)
;

select has_extension('basejump-supabase_test_helpers')
;

select *
from finish()
;
rollback
;
```

**Comment by nfcampos** [comment]
I see. Fair enough. I think this discussion might make more sense once you
support DDL, as there multiple statements per file will be the norm rather
than the exception.

For what it’s worth empty lines are usually good visual separation for me.
You could even enforce it by always adding eg. Two new lines after a
semicolon?

**Comment by nevdelap** [comment]
(Having typed this I've just noticed how old this discussion is.)

I would prefer the semicolons be on the same line since I can put blank lines for the separation myself, but I could maybe get used to it. I do have examples like @elyobo's that look pretty silly and annoying, like in create procedure and function statements.

(Since it allows blank lines, it would be good if it replaced multiple consecutive blank lines with a single one, and instead of having semicolons on their own line why not just `;\n\n` - but its been around a long time, so just a rhetorical question really.)

But what's more problematic is that it doesn't seem to be consistent?

![image](https://github.com/user-attachments/assets/10b22efb-801f-4e69-b894-a02625a28c57)
![image](https://github.com/user-attachments/assets/b53d7632-5383-4ac0-b799-c172be52be78)
